pkgs/applications/audio/musescore/default.nix

```diff
@@ -1,65 +1,140 @@
   pname = "musescore";
-  version = "4.0.2";
+  version = "4.1.0";
 
   src = fetchFromGitHub {
     owner = "musescore";
     repo = "MuseScore";
     rev = "v${version}";
-    sha256 = "sha256-3NSHUdTyAC/WOhkB6yBrqtV3LV4Hl1m3poB3ojtJMfs=";
+    sha256 = "sha256-CqW1f0VsF2lW79L3FY2ev+6FoHLbYOJ9LWHeBlWegeU=";
   };
   patches = [
-    # See https://github.com/musescore/MuseScore/issues/15571
-    (fetchpatch {
-      url = "https://github.com/musescore/MuseScore/commit/365be5dfb7296ebee4677cb74b67c1721bc2cf7b.patch";
-      hash = "sha256-tJ2M21i3geO9OsjUQKNatSXTkJ5U9qMT4RLNdJnyoKw=";
-    })
+    # Fixes the installPhase for darwin
+    ./darwin-dont-install-qml-files.patch
   ];
 
   cmakeFlags = [
-    "-DMUSESCORE_BUILD_CONFIG=release"
-    # Disable the _usage_ of the `/bin/crashpad_handler` utility. See:
-    # https://github.com/musescore/MuseScore/pull/15577
-    "-DBUILD_CRASHPAD_CLIENT=OFF"
+    "-DMUSESCORE_BUILD_MODE=release"
+    # Disable the build and usage of the `/bin/crashpad_handler` utility - it's
+    # not useful on NixOS, see:
+    # https://github.com/musescore/MuseScore/issues/15571
+    "-DMUE_BUILD_CRASHPAD_CLIENT=OFF"
     # Use our freetype
     "-DUSE_SYSTEM_FREETYPE=ON"
   ];
 
   qtWrapperArgs = [
     # MuseScore JACK backend loads libjack at runtime.
-    "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libjack2 ]}"
+    "--prefix ${lib.optionalString stdenv.isDarwin "DY"}LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libjack2 ]}"
+  ] ++ lib.optionals (!stdenv.isDarwin) [
     # There are some issues with using the wayland backend, see:
     # https://musescore.org/en/node/321936
     "--set-default QT_QPA_PLATFORM xcb"
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ninja ];
+  # HACK `propagatedSandboxProfile` does not appear to actually propagate the sandbox profile from `qtbase`
+  sandboxProfile = toString qtbase.__propagatedSandboxProfile or null;
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+    cmake
+    pkg-config
+    ninja
+  ];
 
   buildInputs = [
-    alsa-lib libjack2 freetype lame libogg libpulseaudio libsndfile libvorbis
-    portaudio portmidi flac # tesseract
-    qtbase qtdeclarative qtgraphicaleffects qtquickcontrols2
-    qtscript qtsvg qttools qtwebengine qtxmlpatterns qtnetworkauth qtx11extras
+    libjack2
+    freetype
+    lame
+    libogg
+    libpulseaudio
+    libsndfile
+    libvorbis
+    portaudio'
+    portmidi
+    flac
+    qtbase
+    qtdeclarative
+    qtgraphicaleffects
+    qtquickcontrols
+    qtquickcontrols2
+    qtscript
+    qtsvg
+    qtxmlpatterns
+    qtnetworkauth
+    qtx11extras
+  ] ++ lib.optionals stdenv.isLinux [
+    alsa-lib
   ];
 
+  # Upstream assumes the user has x86_64, and not aarch64
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace build/cmake/SetupBuildEnvironment.cmake \
+      --replace 'set(CMAKE_OSX_ARCHITECTURES x86_64)' 'set(CMAKE_OSX_ARCHITECTURES ${stdenv.hostPlatform.darwinArch})'
```

Comment:
I wonder whether this can be set with a `cmakeFlag` (question for @reckenrode ).

Reply:
There is `BUILD_MACOS_APPLE_SILICON`, which can be set to `ON`. It works by clearing out `CMAKE_OSX_ARCHITECTURES`, which (as I understand it) means to build for the native host only. That should also work normally, but I’m not sure what it would do in a cross build.

I tried cross-building musescore (because x86_64-darwin to aarch64-darwin should work), but it appears waf needs an emulator for an aarch64-darwin cross-build, and there isn’t one available.

Reply:
> There is `BUILD_MACOS_APPLE_SILICON`, which can be set to `ON`. It works by clearing out `CMAKE_OSX_ARCHITECTURES`, which (as I understand it) means to build for the native host only. That should also work normally, but I’m not sure what it would do in a cross build.

Cross compilation is a very high ideal for this package IMO. I removed that `substituteInPlace` in favor of the mentioned cmake flag which I think will survive updates better, especially if upstream will rewrite/rename their cmake files.